custom_components/abbfreeathome_ci/__init__.py

```diff
@@ -166,19 +166,24 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         configuration_url=entry.data[CONF_HOST],
     )
 
-    # Check, if sub-devices for multi-devices should be created
-    if entry.data[CONF_CREATE_SUBDEVICES]:
-        for _device in _free_at_home.get_devices().values():
-            if _device.is_multi_device:
-                device_registry.async_get_or_create(
-                    config_entry_id=entry.entry_id,
-                    identifiers={(DOMAIN, _device.device_serial)},
-                    name=_device.display_name,
-                    manufacturer="ABB Busch-Jaeger",
-                    serial_number=_device.device_serial,
-                    suggested_area=None,
-                    via_device=(DOMAIN, entry.data[CONF_SERIAL]),
-                )
+    for _device in _free_at_home.get_devices().values():
+        # Skip special F@H devices
+        if (
+            _device.device_serial.startswith("FFFF")  # Scenes
+            or _device.device_id == "FFFF"  # System Access Point
+            or len(_device.channels.keys()) == 0
+        ):
+            continue
```

Comment:
I like this, I'm wondering, would looking for the number of channels be enough? If they have a channel they'll likely have an entity.

In Python you can also use `_device.channels` to check if the dict is empty.

```suggestion
        if not _device.channels:
            continue
```

Reply:
No, looking for the number of channels isn't enough. E.g. Scenes always have one channel and the SysAP with a deviceId="FFFF" (serial="ABB700000000") also.

The problem I just faced with (but I think this is a local problem) is, that my configuration shows two "SmartTouch 10" panels. One I'm using (assigned to a room, door-bell event and so on) and another one (with a different serial) also with a lot of channels but not assigned to rooms. [And I don't have and never had two of these panels] This second panel shows up in the new code as an empty device. As stated: I think this is a local problem with my F@H configuration and has nothing to do with the code, but I just want to make some additional tests before I think the code is clean.

Reply:
I see one other scenario that could come up where a device could be added but all of it's channels are orphans. I'll add one additional function to the FreeAtHome class called `get_channels_by_device` which will pull in the filtered channels vs the channels directly from the device.

Reply:
Regarding the ghost device, your not alone. I have 2 Smarttouch 10 and after the last update 2 new appeared with different serials, I deleted them aswell.

Reply:
> No, looking for the number of channels isn't enough. E.g. Scenes always have one channel and the SysAP with a deviceId="FFFF" (serial="ABB700000000") also.
> 
> The problem I just faced with (but I think this is a local problem) is, that my configuration shows two "SmartTouch 10" panels. One I'm using (assigned to a room, door-bell event and so on) and another one (with a different serial) also with a lot of channels but not assigned to rooms. [And I don't have and never had two of these panels] This second panel shows up in the new code as an empty device. As stated: I think this is a local problem with my F@H configuration and has nothing to do with the code, but I just want to make some additional tests before I think the code is clean.

Can you give it another go with the new function I just added `get_channels_by_device`? If it returns any channels for a device then that channel will be added to an entity and will need a device. So I'm wondering what scenario exists that this isn't the case.

I adjusted my local code to use.

```pyhon
        if not _free_at_home.get_channels_by_device(_device.device_serial):
            continue
```

And now all of my devices also have at least 1 entity (e.g. no extra SysAP device)

Reply:
Also to note, I've compared my "live" Home Assistant instance with this branch and my dev instances and the number match-up exactly.

Granted, I don't have any many types of devices. But this is what I'd expect. If you have discrepancies after this change I'd like to know where those are at.